synapse/util/check_dependencies.py

```diff
@@ -66,6 +66,17 @@ def _is_dev_dependency(req: Requirement) -> bool:
     )
 
 
+def _should_ignore_runtime_requirement(req: Requirement) -> bool:
+    # This is a build-time dependency. Irritatingly, `poetry build` ignores the
+    # requirements listed in the [build-system] section of pyproject.toml, so in order
```

Comment:
Should we link to a poetry bug here?

Reply:
https://github.com/python-poetry/poetry/issues/6154 I guess would be this and is linked to someplace else, so probably OK not to.

Reply:
I wasn't 100% sure if this is considered a bug by poetry's authors, but the issue is a good point of reference---I'll gladly add it